rest-api/model/bq_base.py

```diff
@@ -309,22 +338,135 @@ class BQView(object):
   sql = None
 
 
+# class BQSession(object):
+#
+#   def __init__(self, project_id, dataset_id):
+#     self._project_id = project_id
+#     pass
+#
+#   def execute(self, query, page_size=1000):
+#     """
+#
+#     :param query: bigquery sql string
+#     :param project_id: gcp project id
+#     :param dataset_id: bigquery dataset id
+#     :param page_size: maximum number of records per page.
+#     :return: BQRecordSet object
+#     """
+#     pass
+
+
 class BQRecordSet(object):
   """
   Represents a BigQuery data set.
   """
 
   _bq_job = None
 
-  def __int__(self, bq_job):
+  def __int__(self, bigquery_job):
     """
-    :param bq_job: A BigQueryJob object.
+    :param bigquery_job: A BigQueryJob object.
     """
-    self._bq_job = bq_job
+    self._bq_job = bigquery_job
     pass
 
 
-class BQRow(object):
+class BQRecord(object):
   """
-  Represents a single BigQuery data row.
+  Represents a single BigQuery data record.
   """
+  __schema__ = None
+  __fields__ = None
+
+  def __init__(self, schema=None, data=None):
+    """
+    :param schema: BQSchema type, BQSchema instance or schema json string.
+    :param data: initial dict of data for object
+    """
+    if schema:
+      # if schema is a json string, convert it to a BQSchema object.
+      if isinstance(schema, str):
+        self.__schema__ = BQSchema(schema)
+      else:
+        # See if schema is a Class or an object Instance.
+        try:
+          self.__fields__ = schema.get_fields()
+          self.__schema__ = schema
+        except TypeError:
+          self.__schema__ = schema()
+      self.__fields__ = self.__schema__.get_fields()
+
+    if data:
+      self.update_values(data)
+
+  def __getitem__(self, item):
+    """
+    Lookup item data value
+    :param item: string
+    :return: value
+    """
+    return getattr(self, item)
+
+  def update_values(self, data):
+    """
+    Update this object with the given dict values, validate against BQSchema if available.
+    :param data: dict of data values to add/update.
+    """
+    def update(d, u, s):
```

Comment:
what does d/u/s represent ?

Reply:
d=dict to update
u=dict to update from
s=schema